File: src/main/java/my/study/repository/design/pattern/chain/ConcreteHandlerC.java
package my.study.repository.design.pattern.chain;

/**
 * @author djh on  2018/9/16 20:24
 * @E-Mail 1544579459@qq.com
 */
class ConcreteHandlerC extends AbstractHandler {


    @Override
    void handlerRequest(int money) {
        System.out.println("终极处理者,买到宝马7系,恭喜发财!!");
    }
}

File: src/main/java/my/study/repository/design/pattern/chain/kotlin/ConcreteHandlerC.kt
package my.study.repository.design.pattern.chain.kotlin

/**
 * @author djh on  2018/10/4 18:26
 * @E-Mail 1544579459@qq.com
 */
class ConcreteHandlerC : AbstractHandler() {
    override fun handlerRequest(money: Int) {
        println("总结处理者, $money 万, 买宝马7系,恭喜发财!")
    }
}